src/lib.rs

```diff
@@ -0,0 +1,1607 @@
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use rstd::prelude::*;
+
+use codec::{Codec, Decode, Encode};
+use runtime_primitives::traits::{MaybeSerialize, Member, One, SimpleArithmetic};
+use srml_support::traits::Currency;
+use srml_support::{decl_module, decl_storage, ensure, Parameter};
+
+use rstd::iter::Iterator;
+
+use runtime_primitives::traits::Zero;
+
+use crate::sr_api_hidden_includes_decl_storage::hidden_include::traits::Imbalance;
+
+use rstd::collections::btree_map::BTreeMap;
+use rstd::collections::btree_set::BTreeSet;
+
+mod hiring;
+mod macroes;
+mod mock;
+mod test;
+
+pub use hiring::*;
+use system;
+
+use stake;
+
+/// ...
+pub trait ApplicationDeactivatedHandler<T: Trait> {
+    /// An application, with the given id, was fully deactivated, with the
+    /// given cause, and was put in the inactive state.
+    fn deactivated(application_id: &T::ApplicationId, cause: &hiring::ApplicationDeactivationCause);
+}
+
+pub trait Trait: system::Trait + stake::Trait + Sized {
+    type OpeningId: Parameter
+        + Member
+        + SimpleArithmetic
+        + Codec
+        + Default
+        + Copy
+        + MaybeSerialize
+        + PartialEq;
+
+    type ApplicationId: Parameter
+        + Member
+        + SimpleArithmetic
+        + Codec
+        + Default
+        + Copy
+        + MaybeSerialize
+        + PartialEq;
+
+    /// Type that will handle various staking events
+    type ApplicationDeactivatedHandler: ApplicationDeactivatedHandler<Self>;
+}
+
+/// Helper implementation so we can provide multiple handlers by grouping handlers in tuple pairs.
+/// For example for three handlers, A, B and C we can set the StakingEventHandler type on the trait to:
+/// type StakingEventHandler = ((A, B), C)
+///
+impl<T: Trait> ApplicationDeactivatedHandler<T> for () {
+    fn deactivated(
+        _application_id: &T::ApplicationId,
+        _cause: &hiring::ApplicationDeactivationCause,
+    ) {
+    }
+}
+
+/*
+impl<T: Trait, X: ApplicationDeactivatedHandler<T>, Y: ApplicationDeactivatedHandler<T>> ApplicationDeactivatedHandler<T>
+    for (X, Y)
+{
+    fn deactivated(
+        application_id: T::ApplicationId,
+        cause: hiring::ApplicationDeactivationCause
+    ) {
+        X::deactivated(application_id, cause);
+        Y::deactivated(application_id, cause);
+    }
+}
+*/
+
+pub type BalanceOf<T> =
```

Comment:
I agree, but the better solution would perhaps be to put this alias in one of those util package? 

https://github.com/Joystream/substrate-utils

It's used all over the place, also in modules that don't have `stake` module as a dependency, and they keep redefining it.

Reply:
I'm not sure this is an alias that can just be factored out into a utils package because its really depends on how you setup the currencies in the runtime. (In our case we will only have a single currency anyway so maybe its a moot point) But I think for now lets just leave it as is.